src/mpyl/steps/deploy/__init__.py

```diff
@@ -1,4 +1,37 @@
 """
 Step implementations relating to the `Deploy` Stage
 """
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional
+
+from ...project import Project, load_project
+from ...stages.discovery import find_invalidated_projects_for_stage
+from ...utilities.repo import RepoConfig, Repository
+
 STAGE_NAME = "deploy"
+
+
+@dataclass(frozen=True)
+class DeploySet:
+    all_projects: set[Project]
+    projects_to_deploy: set[Project]
+
+
+def find_deploy_set(repo_config: RepoConfig, tag: Optional[str]) -> DeploySet:
```

Comment:
Please keep spamming me for reviews!

Reply:
`find_deploy_set` had a bug with manual selection. We found that the core of the problem was in the code duplication, so we decided to move the build/deploy set into the run_props so it would only have to be constructed once at the beginning of a run and could use the same code. This also fixed the bug of course.
We did add you as a reviewer and waited a few days but like I said before: we didn't want to spam you on whatsapp or anything.